File: PSET2/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{

    // Get the text

    string text = get_string("Text: ");

    float letters = 0;
    float words = 1;
    float sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {

        // Calculate the number of letters
        if ((text[i] >= 97 && text[i] <= 122) || (text[i] >= 65 && text[i] <= 90))
        {
            letters++;
        };

        // Calculate the number of words
        if (text[i] == 32)
        {
            words++;
        };

        // Calculate the number of sentences
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        };
    };

    float L = 100 * (letters / words);
    float S = 100 * (sentences / words);

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Check the result

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    };
}

File: PSET6/sentimental-readability/readability.py
import string
from cs50 import get_string

lett = 0
senten = 0
content = get_string("Text: ")

for j in content:
    if (j.isalpha()):
        lett += 1

word = len(content.split())

for l in content:
    if (l == '.' or l == '?' or l == '!'):
        senten += 1

wordsinsentence = word / 100

s = senten / wordsinsentence
l = lett / wordsinsentence

indx = round(0.0588 * l - 0.296 * s - 15.8)

if indx < 1 :
    print("Before Grade 1")
elif indx >= 16:
    print("Grade 16+")
else:
    print(f"Grade {indx}")
